addons/mail/models/mail_thread.py

```diff
@@ -2135,6 +2135,41 @@ def _message_create(self, values_list):
     # NOTIFICATION API
     # ------------------------------------------------------
 
+    @api.model
+    def discard_notifications(self, notification_type):
+        if notification_type in {'email', 'sms'}:
+            # TDE CHECK: delete pending SMS
+            author_id = self.env.user.partner_id.id
+            self._cr.execute("""
+                            SELECT notif.id, mes.id
+                            FROM mail_notification notif
+                            JOIN mail_message mes
+                                ON notif.mail_message_id = mes.id
+                            WHERE notif.notification_type = %(notification_type)s IS TRUE 
+                              AND notif.author_id = %(author_id)s
+                              AND notif.notification_status IN ('bounce', 'exception')
+                              AND mes.model = %(model_name)s
+                                """,
+                             {'model_name': self._name, 'author_id': author_id, 'notification_type': notification_type})
+            res = self._cr.fetchall()
+            notif_ids = [row[0] for row in res]
```

Comment:
Could be simplified using zip, like
```
notif_ids, msg_ids = zip(*self._cr.fetchall())
```
This should unzip the list of tuples into two lists.